samples/add-feature-collection-layer-from-table/src/main/java/com/esri/arcgismaps/sample/addfeaturecollectionlayerfromtable/components/AddFeatureCollectionLayerFromTableViewModel.kt

```diff
@@ -0,0 +1,222 @@
+/* Copyright 2025 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.addfeaturecollectionlayerfromtable.components
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.Color
+import com.arcgismaps.data.FeatureCollection
+import com.arcgismaps.data.FeatureCollectionTable
+import com.arcgismaps.data.Field
+import com.arcgismaps.data.FieldType
+import com.arcgismaps.geometry.Polygon
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.geometry.Polyline
+import com.arcgismaps.geometry.SpatialReference
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.layers.FeatureCollectionLayer
+import com.arcgismaps.mapping.symbology.SimpleFillSymbol
+import com.arcgismaps.mapping.symbology.SimpleFillSymbolStyle
+import com.arcgismaps.mapping.symbology.SimpleLineSymbol
+import com.arcgismaps.mapping.symbology.SimpleLineSymbolStyle
+import com.arcgismaps.mapping.symbology.SimpleMarkerSymbol
+import com.arcgismaps.mapping.symbology.SimpleMarkerSymbolStyle
+import com.arcgismaps.mapping.symbology.SimpleRenderer
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.launch
+
+/**
+ * ViewModel which creates three in-memory FeatureCollectionTables (point, polyline, polygon),
+ * adds features to them, creates a FeatureCollection and displays it on the map as a
+ * FeatureCollectionLayer.
+ */
+class AddFeatureCollectionLayerFromTableViewModel(app: Application) : AndroidViewModel(app) {
+
+    // ArcGIS map exposed for Compose screen.
+    val arcGISMap = ArcGISMap(BasemapStyle.ArcGISOceans).apply {
+        initialViewpoint = Viewpoint(latitude = 8.849289, longitude = -79.497238, scale = 1e6)
+    }
+
+    // Message dialog view model for error handling
+    val messageDialogVM = MessageDialogViewModel()
+
+    init {
+        // Load the map and then build the feature collection layer
+        viewModelScope.launch {
+            arcGISMap.load()
+                .onFailure { messageDialogVM.showMessageDialog(it) }
+                .onSuccess {
+                    // Create feature collection tables and add them to the map
+                    createAndAddFeatureCollectionLayer()
+                }
```

Comment:
Using above change, you could use a sequential load in the launched scope:
```suggestion
            // Create feature collection tables and add them to the map
            createAndAddFeatureCollectionLayer()
            arcGISMap.load().onFailure { messageDialogVM.showMessageDialog(it) }
```

Reply:
modified